portals/devportal/src/main/webapp/services/login/login_callback.jsp

```diff
@@ -126,16 +126,30 @@
         String tokenEndpoint = Util.getLoopbackOrigin((String) Util.readJsonObj(settings, "app.origin.host")) + TOKEN_URL_SUFFIX;
         String data = "code=" + request.getParameter("code") + "&grant_type=authorization_code&redirect_uri=" + loginCallbackUrl;
         String codeVerifier = (String) session.getAttribute("code_verifier");
+        boolean isBypassClientCredentials = systemApplicationDAO.isBypassClientCredentials(clientId);
         if (codeVerifier != null) {
             data = data + "&code_verifier=" + codeVerifier;
-        }
+            if (isBypassClientCredentials) {
+                data = data + "&client_id=" + clientId;
+            }
+        }  
         HttpClient client = HttpClient.newHttpClient();
-        HttpRequest post = HttpRequest.newBuilder()
+        HttpRequest post;
+        if (isBypassClientCredentials && codeVerifier != null) {
+            post = HttpRequest.newBuilder()
+                .uri(URI.create(tokenEndpoint))
+                .POST(HttpRequest.BodyPublishers.ofString(data))
+                .header("Content-Type", "application/x-www-form-urlencoded")
+                .build();
+        }
+        else {
```

Comment:
formatting issue. else should come one line above

Reply:
Fixed by https://github.com/wso2/apim-apps/pull/469/commits/52f9d9eaf061f2e1037b499257e82cb46dd7d4d6